alphalens/performance.py

```diff
@@ -527,3 +529,109 @@ def average_cumulative_return(q_fact, demean_by):
+              .fit()
+              .rsquared)
+        vifs[i] = 1/(1-r2)
+    return pd.Series(index=data.columns, data=vifs)
+
+
+def decompose_returns(algo_returns, risk_factors, hierarchy=None):
+    '''
+    Decomposes returns into returns attributable to common risk factors.
+    Parameters
+    ----------
+    algo_returns : pd.Series
+        Algorithm's daily returns, with a tz-localized index
+        - Example:
+        2010-01-01 00:00:00+00:00    0.812385
+        2010-01-04 00:00:00+00:00    0.468386
+        2010-01-05 00:00:00+00:00    0.680892
+    risk_factors : pd.DataFrame
+        Daily common risk factors. Index is tz-localized DatetimeIndex,
+        columns are common factor names
+        - Example:
+                                        Mkt       SMB       HML       UMD
+        2017-06-26 00:00:00+00:00    0.0004   -0.0008    0.0069   -0.0057
+        2017-06-27 00:00:00+00:00   -0.0084   -0.0043    0.0130   -0.0030
+        2017-06-28 00:00:00+00:00    0.0102    0.0077    0.0021    0.0091
+    hierarchy : OrderedDict
+        Hierarchy of common factors, for use in plotting the bar chart.
+        If None, defaults to a hierarchy in which each column of risk_factors
+        forms a 'level' of the hierarchy.
+        - Example:
+        OrderedDict([('Market', ['Mkt']),
+                     ('Style', ['SMB', 'HML', 'UMD'])])
+    Returns
+    -------
+    returns_decomposition : pd.DataFrame
+        Decomposed returns
+    betas : pd.DataFrame
+        Betas to the common factors
+    t_stats : pd.DataFrame
+        t-statistics for the betas to the common factors. Computed by dividing
+        betas by White's (1980) heteroskedasticity standard errors.
+    vifs : pd.Series
+        Variance inflation factors for the common factors
+        - For more information, see
+        https://en.wikipedia.org/wiki/Variance_inflation_factor
+    '''
+    if hierarchy is None:
+        hierarchy = risk_factors.columns.values
+
+    # construct a cumulative hierarchy
+    cum_tiers = []
+    for i in range(len(hierarchy.values())):
+        cum_tiers.append([item for a in range(0, i+1)
+                          for item in list(hierarchy.values())[a]])
+    cum_hierarchy = OrderedDict()
+    cum_hierarchy.update((list(hierarchy.keys())[i], cum_tiers[i])
+                         for i in range(len(hierarchy)))
+
+    # ensure that risk_factors and algo_rets_over_rf have the same index
+    idx = algo_returns.index.intersection(risk_factors.index)
+    risk_factors = risk_factors.loc[idx]
+    algo_returns = algo_returns.loc[idx]
+
+    idx = np.append(['Alpha'], risk_factors.columns.values)
+    betas = pd.DataFrame(index=idx, columns=hierarchy.keys())
+    returns_decomposition = pd.DataFrame(index=idx, columns=hierarchy.keys())
+    t_stats = pd.DataFrame(index=idx, columns=hierarchy.keys())
+
+    for tier, factors in cum_hierarchy.iteritems():
```

Comment:
I think we can get away with changing `.iteritems()` to `.items()`. This operation seems to be bounded by the number of risk factors, which should usually be a reasonable number.